addons/mail/controllers/discuss.py

```diff
@@ -107,3 +107,7 @@ def message_get_suggested_recipients(self, model, res_ids):
         except Exception:
             return {}
         return records._message_get_suggested_recipients()
+
+    @http.route('/mail/edit_message/<model("mail.message"):message>', type='json', auth='user')
```

Comment:
Hello,

Standard controller naming we try to enforce would be more like ``/message/<int:mail_message_id>/edit_body``, aka: module or model prefix (mail_message, message ... would avoid generic 'mail' if possible that is more for end user controllers, this one is quite technical), record (here an ID as slug is not required for beautification), then action.

Reply:
I strongly oppose this idea. This makes searching for those routes in the code extremely complex as there is a dynamic part in the middle of it. I've had the pain during review of the guest PR and I plan to change them all.
It's also making the tests and the mock server much harder to write as we need to use regex to match urls instead of just simple string comparison.

Reply:
That's the standard naming decided with Antony: model then action. It is not officially in guidelines but we are going to move all routes gradually to this, like ``channel/<channel_id>/edit`` . Grepping is easy as you have to grep 'channel.*edit', which is not that hard.

Reply:
Rationale is to have model-based routes, and not based on action itself. Model is more important than the action we do on it.

Reply:
A lot of actions are done on multiple records at the same time (eg. all the actions from list view when selecting multiple records, mark as read for multiple messages at once, etc).
Some action also concern several different type of records at the same time (eg. the "is typing" action is for a specific user on a specific channel, both are equally necessary). 
It's the action that is important, the record(s) is(are) just a param of it.

Reply:
@seb-odoo we use the /mail prefix on routes in https://github.com/odoo/odoo/pull/75496 so please let me know if you want us to keep it or not. Also please clarify whether we do want the id in the URL or as a POST param.

Reply:
So just I think there should be no enforced rule for technical/POST routes, and I'm against the suggested guideline for them.

But I can understand why we would want a guideline for the SEO relevant routes that are used on website, and for them I'm fine with your format if this proved to be the most effective one.

Reply:
Actually most odoo routes are just for a single record as they are used for portal / frontend display. Or they are generic (display events, display channels, display portal invoices). We want to have routes like folders: "event" domain, then go into event "Youpie Event" 3, then perform action "publish", that would be ``/event/<event_slug>/publish`` .

But indeed you have some controllers specific to JS widgets which work in batch. In that case that's like a record set but as this is not supported, naming would effectively skip the record and concatenate model and action like ``/model/action?record_ids=[]`` .

But you have to know that all routes are going to move to that direction in our apps when possible. Obviously if you want to do your own popotte in discuss, that's just a suggestion.

Reply:
> So just I think there should be no enforced rule for technical/POST routes, and I'm against the suggested guideline for them.
> 
> But I can understand why we would want a guideline for the SEO relevant routes that are used on website, and for them I'm fine with your format if this proved to be the most effective one.

Sorry, actually this is a JSON route. Indeed for JSON (actually mostly technical), guidelines would be ``/mail/edit?message_id=3``, with message_id being a simple parameter of the route.

cc @sdegueldre

Reply:
For prefix:
Last I checked with AL he preferred to have them called `/mail` while we are in the `mail` module, also for message edit we are actually editing a "mail.message" here so I think that makes sense to still have mail on it no?

For some other routes if we allow ourselves to not follow the "module prefix" I'd just like them all to start with `/discuss` I guess? But I don't think it applies here as this is one of the few actual common route mail/discuss.

Reply:
I would say ``/mail`` or ``/message`` . Actually purpose would be to prefix by the model more than the module when it gives more information. Here I think both are understandable. ``/discuss`` is not really a model, but we could think of moving all routes specific to discuss into a discuss sub-naming to better highlight that they should not be used outside of this scope. In the end what matters is to have a common guideline that helps having some consistency, but also understandable code :D .